core/src/transform/desugar_destructuring.rs

```diff
@@ -1,222 +1,203 @@
-//! Desugar destructuring
+//! Destructuring desugaring
 //!
-//! Replace a let-binding with destructuring by a classical let-binding.
-//! It will first destruct the pattern and create a new var for each field of the pattern.
-//! After that, it will construct a new Record/Array from the extracted fields.
+//! Replace a let-binding with destructuring by a sequence of normal let-binding.
 //!
 //! # Example
 //!
-//! ## The let pattern:
+//! ## Let-binding
+//!
+//! The following destruring let-binding:
+//!
 //! ```text
 //! let x @ {a, b=d, ..} = {a=1,b=2,c="ignored"} in ...
 //! ```
+//!
 //! will be transformed to:
+//!
 //! ```text
 //! let x = {a=1,b=2,c="ignored"} in
 //! let a = x.a in
 //! let d = x.b in
 //! ...
 //! ```
 //!
-//! ## The function pattern
+//! ## Function
+//!
+//! The following desctructuring function:
```

Comment:
```suggestion
//! The following destructuring function:
```

Reply:
It's definitely hard to type, or so it seems :laughing: